core/services/job/spawner.go

```diff
@@ -240,9 +241,17 @@ func (js *spawner) DeleteJob(ctx context.Context, jobID int32) error {
 
 	aj.delegate.BeforeJobDeleted(aj.spec)
 
-	ctx, cancel := utils.CombinedContext(js.chStop, ctx)
+	combctx, cancel := utils.CombinedContext(js.chStop, ctx)
 	defer cancel()
-	err := js.orm.DeleteJob(ctx, jobID)
+
+	// Inject the wrapping transaction into the combined context if it exists.
+	// This allows rollbacks from other services.
+	tx := postgres.GetTxFromContext(ctx)
+	if tx != nil {
+		combctx = postgres.InjectTxIntoContext(combctx, tx)
```

Comment:
@archseer what is your opinion on this approach? Does it mesh well with the work you've done on DB stuff?

Reply:
I personally prefer passing the `tx`/`db` into the function rather than hiding it in a context, I use a interface to be able to pass either (and don't use the transaction manager at all): https://github.com/smartcontractkit/chainlink/blob/58a5e1fce6be6ba077e26541e820592c0ad310a9/core/services/postgres/sqlx.go#L16-L20

Let's merge this as is for now, since it's a larger discussion.